src/main/java/software/amazon/cloudformation/resource/IdentifierUtils.java

```diff
@@ -66,4 +76,85 @@ public static String generateResourceIdentifier(final String logicalResourceId,
         return sb.append(RandomStringUtils.random(GUID_LENGTH, 0, 0, true, true, null, new Random(clientRequestToken.hashCode())))
             .toString();
     }
+
+    public static String generateResourceIdentifier(final String stackId,
+                                                    final String logicalResourceId,
+                                                    final String clientRequestToken,
+                                                    final int maxLength) {
+
+        String stackName = stackId;
+
+        if (isStackPortionValid(stackId)) {
+            List<String> stackParts = ImmutableList.copyOf(STACKID_SPLITTER.split(stackId));
```

Comment:
Why need to copy to a ImmutableList `stackParts`? can be simple as `STACKID_SPLITTER.split(stackId)[1]`?

Reply:
yep